### PR TITLE
feat: Naxx lvl60, naxx25 with autoscaler and DB changes

### PR DESCRIPTION
**Pull request by ghost**
I've been having some fun implementing naxx lvl 60. I'm aware naxx lvl60 was removed in 3.0.x and shouldn't be part of WOTLK progression module, but most data is present in AzerothCore DB so I figured I could give it a try and see the result.

Dev document with pictures/video more details https://github.com/SoglaHash/mod-progression-system-naxx25-60/blob/naxx/README_naxx.md

# Result

I achieved a lvl 60 Naxx based on Naxxramas25 mechanics and an easy scaling option set in config to tune the difficulty. As of now the hp/dmg with `1.0f` scaling is very similar to 40man-naxx.

I updated the teleporting to be in EPL incl. graveyard, summon stone and attunement requirements. The original ziggurath is gone so I improvised with an orb gameobject that requires attunement to use.

Added the floating naxxramas transport object in EPL 

The T3 quests, adjustment to echoes of war, frost resistance quests, archmage and trainer (Omarion) is present

# How
TLDR; I use a [stripped](https://github.com/azerothcore/mod-autobalance/compare/master...SoglaHash:mod-autobalance-naxx25-60:clean?expand=1) `autoscaler`  that only modifies Naxxramas creatures. Then to the DB I added the lvl 60 loot, objects, quests that are missing. 

[Dev doc](https://github.com/SoglaHash/mod-progression-system-naxx25-60/blob/naxx/README_naxx.md) contains details 
# PROS/CONS

The big advantage is that it re-used the Naxxramas scripts already present. Downside is that as values are overwritten, lvl80 and lvl60 can't co-exist. 

I use a cheap hack  to only scale Naxx mobs but the script hooks to all creatures.
```
if (creature->GetMapId() != 533)
    return;
```

Some blizzlike features aren't present such as the original Naxx40 teleporter (ziggurath), no portals to outside (areatriggers can be disabled). 

# Issues/TODOs

Require the blizzlike blue portal to activate fixes to ACore: https://github.com/azerothcore/azerothcore-wotlk/pull/12696 
I left the blue naxxramas orbs in for now, but can be removed easily

need to set 25man difficulty to enter 
```
/script SetRaidDifficulty(2)
```  

Players can be summoned into the instance without attunement, but require attunement completed to use transporter, boss portals and access to T3/frost resistance quests.

Scourge Event is missing, it is present in Cmangos/Vmangos but requires quite a bit of rewriting to make it work. The creatures/objects are present in but missing scripting.

## SOURCE:
most of the data is already in ACore and just needs linking

used CMangos/VMangos DB when values were missing

### CODE REVIEW

**Comment by heyitsbench** [comment]
I am personally of the opinion that the scourge event should be added to the main AC repo, both the vanilla and Wrath versions, as separate events, rather than stuck in this module, but that's just me.

**Comment by BarbzYHOOL** [comment]
reading the desc, looks great

**Comment by howdya** [comment]
This is great. Do we have any update? 8 months already.

**Comment by Fer0Fer0** [comment]
Anybody still working on this PR?
I am willing to offer a 200$+ bounty to anybody who is able to finalize this.
From what i can gather there are still a few things left to do, as well as ensure that encounters are working properly.

**Comment by heyitsbench** [comment]
Question: any idea of how much is left to do/correct?

**Comment by sogladev** [comment]
I forgot about this PR, I abandoned it and moved scripts/SQL into a separate module https://github.com/sogladev/mod-vanilla-naxxramas 

I'd say it's ready for testing. Most stuff is done. One big todo is looking at every single physical damage ability.  I've been putting it off as it's a chore and can't come up with a good approach ; ```non-spell, physical+cleave+weapon damage scaling. Every creature_template added in naxx40_creatures.sql might need their DamageModifier adjusted to reduce dealt damage```

Additionally, I need a look at *cpp scripts and do refactors to make it easier to maintain, like moving the spell scripts into a seperate file

**Comment by 55Honey** [comment]
Thanks for sharing @sogladev !
